src/polyswarmd/bounties.py

```diff
@@ -51,6 +54,36 @@ def calculate_commitment(account, verdicts):
     return int_from_bytes(nonce), int_from_bytes(commitment)
 
 
+def substitute_ipfs_metadata(session, config, ipfs_uri):
+    """Download metadata from IPFS and validate it against the schema.
+
+    :param session: Requests session
+    :param config: Flask config object
+    :param ipfs_uri: Porential IPFS uri string
+    :return: Metadata from IPFS, or original metadata
+    """
+    if not is_valid_ipfshash(ipfs_uri):
+        return ipfs_uri
+
+    future = session.get(config.ipfs_uri + '/api/v0/cat', params={'arg': ipfs_uri}, timeout=1)
+    try:
+        r = future.result()
+        if r.status_code // 100 != 2:
+            logger.critical(f'Got {r.status_code} from ipfs with uri {ipfs_uri}')
+            return ipfs_uri
+
+        content = json.loads(r.text)
+        return content if AssertionMetadata.validate(content) else ipfs_uri
```

Comment:
to make this less complex..
1. remove the return ipfs_uri and the else ipfs_uri. Let those situations get caught outside of the try/except by the bottom return statement
2. make the if r.status_code into an if/else to test what needs testing and do the logging.
3. make the return content if Ass... into an if statement w/o an else. So the else part is caught outside the try block